File: src/contests/cg/puzzle/cpp/google-interview---the-two-egg-problem.cpp
// 🎮 CodinGame Puzzle - google-interview---the-two-egg-problem
// https://www.codingame.com/training/hard/google-interview---the-two-egg-problem

#include <iostream>
using namespace std;

int main() {
    long long n;
    cin >> n;
    long long k = 0;
    while (k * (k + 1) / 2 < n) k++;
    cout << k << endl;
}

File: src/contests/cg/puzzle/dart/google-interview---the-two-egg-problem.dart
// 🎮 CodinGame Puzzle - google-interview---the-two-egg-problem
// https://www.codingame.com/training/hard/google-interview---the-two-egg-problem

import 'dart:io';

void main() {
  int n = int.parse(stdin.readLineSync()!.trim());
  int k = 0;
  while (k * (k + 1) ~/ 2 < n) k++;
  print(k);
}
